### PR TITLE
update cargo-readme

### PR DESCRIPTION
**Pull request by webern**
**Issue number:**

N/A
Related to https://github.com/webern/cargo-readme/pull/84

**Description of changes:**

Update Cargo readme to update its dependencies.

**Testing done:**

If the build works, we're good.

**TODO**

- [x] Publish to crates.io and use that

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: tools/deny.toml [hunk 1]
```diff
@@ -62,7 +62,7 @@ license-files = [
 [bans]
 # Deny multiple versions or wildcard dependencies.
 multiple-versions = "deny"
-wildcards = "deny"
+# wildcards = "deny"
```
Comment (webern) [inline tools/deny.toml]: undo this

File: tools/deny.toml [hunk 2]
```diff
@@ -89,13 +95,13 @@ skip-tree = [
...
     # that differ in major version.
     { name = "windows-sys" },
-    # generate-readme uses an old version of clap and other dependencies
-    { name = "generate-readme", version = "0.1.0" }
 ]
 
 [sources]
 allow-git = [
     "https://github.com/bottlerocket-os/bottlerocket-test-system",
+    # TODO - remove from this PR
```
Comment (webern) [inline tools/deny.toml]: undo this

File: sources/deny.toml [hunk 3]
```diff
@@ -54,32 +54,31 @@ license-files = [
...
     # dependency tree because windows-sys has many sub-crates
     # that differ in major version.
     { name = "windows-sys", version = "=0.42.0" },
-    # generate-readme pulls in an older clap that causes some duplicate
-    # dependencies
-    { name = "generate-readme", version = "=0.1.0" }
 ]
 
 [sources]
+# TODO - temporary. remove from this PR
```
Comment (webern) [inline sources/deny.toml]: undo this

File: sources/generate-readme/Cargo.toml [hunk 4]
```diff
@@ -7,5 +7,6 @@ edition = "2021"
 publish = false
 
 [dependencies]
-cargo-readme = "3"
+# TODO - use crates.io release
```
Comment (webern) [inline sources/generate-readme/Cargo.toml]: fix this

**Comment by webern** [comment]
@ecpullen This function might be gone from `toml`, is this a big problem? I looked around and I don't see it in docs.rs or anything that sounds similar.

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/bottlerocket/compare/e2395c1d2900742e1c117e017171ef3e002ce58c..0d48e3797ba19d8997f0b0f115a29573e13d791b

Use published version

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/bottlerocket/compare/0d48e3797ba19d8997f0b0f115a29573e13d791b..f466ff98dcc919c998755c611842c3beae39c0f5

Attempt a rebase with conflict resolution

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/bottlerocket/compare/f466ff98dcc919c998755c611842c3beae39c0f5..db6d5660c3c34c1b1ae7cfbacf59e31e9b93fa89

Whitespace cleanup

**Comment by webern** [comment]
> https://github.com/bottlerocket-os/bottlerocket/compare/db6d5660c3c34c1b1ae7cfbacf59e31e9b93fa89..ea12bdc888e8c4bf38a07cab832ccf0ff35b71e0

Rebase

File: sources/deny.toml [hunk 5]
```diff
@@ -56,12 +56,16 @@ license-files = [
...
-deny = [
-    { name = "structopt" },
-    { name = "clap", wrappers = [ "cargo-readme" ] },
-]
-
 skip = [
+    # Tough is using an older version of globset which is using
+    # an older version of aho-corasick
+    { name = "aho-corasick", version = "=0.7" },
+    # Netdog is using a feature of toml 0.5 that does not exist in
```
Comment (webern) [inline sources/deny.toml:63]: @zmrow and @yeazelm :cry: 

https://github.com/bottlerocket-os/bottlerocket/issues/3422

**Comment by jpculp** [comment]
I went ahead and bumped `cargo-readme` in #3532, but without bumping `toml` elsewhere since I don't think it will impact the binaries themselves. I suspect we'll want to use this PR to move everything else to `toml 0.7` at the same time.

**Comment by webern** [comment]
I'm going to close this as it's not clear when I will get back to it. Sorry.

**Pull request closed without merging**